platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/BuildNameProvider.kt

```diff
@@ -39,12 +38,19 @@ class BuildNameProvider(private val listenerManager: ListenerManager) : Internal
         listenerManager.removeListener(this)
     }
 
-    var buildName: String? = null
-        get() = field
+    private
+    lateinit var rootBuildName: String
+
+    fun buildName(): String {
+        if (this::rootBuildName.isInitialized) {
```

Comment:
❓  Is there a reflection cost for this?

I think having a nullable private field and checking for null should be enough here.
We can also do the following to have the right nullability expectations:
```
rootBuildName = settings.rootProject.name!!
```